src/rust/src/pkcs7.rs

```diff
@@ -290,11 +296,101 @@ fn smime_canonicalize(data: &[u8], text_mode: bool) -> (Cow<'_, [u8]>, Cow<'_, [
     }
 }
 
+#[cfg(not(CRYPTOGRAPHY_IS_BORINGSSL))]
+fn load_pkcs7_certificates(
+    py: pyo3::Python<'_>,
+    pkcs7: Pkcs7,
+) -> CryptographyResult<&pyo3::types::PyList> {
+    let nid = pkcs7.type_().map(|t| t.nid());
+    if nid != Some(openssl::nid::Nid::PKCS7_SIGNED) {
+        let nid_string = nid.map_or("empty".to_string(), |n| n.as_raw().to_string());
+        return Err(CryptographyError::from(
+            exceptions::UnsupportedAlgorithm::new_err((
+                format!("Only basic signed structures are currently supported. NID for this data was {}", nid_string),
+                exceptions::Reasons::UNSUPPORTED_SERIALIZATION,
+            )),
+        ));
+    }
+
+    let signed_certificates = pkcs7.signed().and_then(|x| x.certificates());
+    match signed_certificates {
+        None => Err(CryptographyError::from(
+            pyo3::exceptions::PyValueError::new_err(
+                "The provided PKCS7 has no certificate data, but a cert loading method was called.",
+            ),
+        )),
+        Some(certificates) => {
+            let result = pyo3::types::PyList::empty(py);
+            for c in certificates {
+                result.append(
+                    load_der_x509_certificate(
+                        py,
+                        pyo3::types::PyBytes::new(py, c.to_der()?.as_slice()).into_py(py),
+                        None,
+                    )?
+                    .into_py(py),
+                )?;
```

Comment:
You're getting coverage issues here. Best resolution is to rewrite as:

```
let cert_der = pyo3::types::PyBytes::new(py, c.to_der()?.as_slice()).into_py(py);
let cert = load_der_x509_certificate(py, cert_der, None)?;
result.append(cert.into_py(py))?;
```

Reply:
thx! fixed